src/test/java/picard/illumina/IlluminaBasecallsToFastqTest.java

```diff
@@ -150,19 +149,21 @@ public void testHiseqxSingleLocs() throws Exception {
         runStandardTest(1, "hiseqxSingleLocs.", "barcode_double.params", 2, "25T8B8B25T",TEST_DATA_HISEQX_SINGLE_LOCS, HISEQX_TEST_DATA_DIR);
     }
 
-    private void compareFastqs(File testDataDir, File outputSam, String filename) {
-        File f1 = new File(outputSam.getParentFile(), filename);
-        File f2 = new File(testDataDir, filename);
-        FastqReader reader1 = new FastqReader(f1);
-        List<FastqRecord> reads = new ArrayList<>();
-        FastqReader reader2 = new FastqReader(f2);
-        for (FastqRecord record : reader1) {
-            reads.add(record);
-        }
+    private void compareFastqs(File actual, File expected) {
+        List<FastqRecord> actualReads = slurpReads(actual);
+        List<FastqRecord> expectedReads = slurpReads(expected);
 
-        for (FastqRecord record : reader2) {
-            Assert.assertTrue(reads.contains(record));
-        }
+        actualReads.sort((a, b) -> a.getReadName().compareTo(b.getReadName()));
+        expectedReads.sort((a, b) -> a.getReadName().compareTo(b.getReadName()));
+        Assert.assertEquals(actualReads, expectedReads);
+    }
+
+    private List<FastqRecord> slurpReads(final File f) {
+        final FastqReader in = new FastqReader(f);
+        final List<FastqRecord> recs = new ArrayList<>();
+        for (final FastqRecord r : in) recs.add(r);
```

Comment:
split up with '{' please
```suggestion
        for (final FastqRecord r : in) {
          recs.add(r);
        }
```